### PR TITLE
Split Go lib test dependencies into separate module

### PR DESCRIPTION
**Pull request by gj**
Tested
- on Create Release matrix in CI:
  - https://github.com/osohq/oso/actions/runs/1737888019
  - https://github.com/osohq/oso/actions/runs/1737800644
  - https://github.com/osohq/oso/actions/runs/1735341101
- and locally on Go 1.14, 1.15, 1.16, 1.17 on macOS
- and by running the Go examples in our docs off of the updated (local) Go lib

### CODE REVIEW

File: languages/go/go.mod [hunk 1]
```diff
@@ -5,16 +5,10 @@ go 1.14
...
 	golang.org/x/crypto v0.0.0-20210921155107-089bfa567519 // indirect
-	golang.org/x/mod v0.5.1 // indirect
-	golang.org/x/net v0.0.0-20211020060615-d418f374d309 // indirect
 	golang.org/x/sys v0.0.0-20211020174200-9d6173849985 // indirect
-	golang.org/x/text v0.3.7 // indirect
-	golang.org/x/tools v0.1.7 // indirect
-	google.golang.org/appengine v1.6.7 // indirect
-	gorm.io/driver/sqlite v1.2.6 // indirect
-	gorm.io/gorm v1.22.5 // indirect
+	gorm.io/driver/sqlite v1.2.6
```
Comment (samscott89) [inline languages/go/go.mod:12]: So will installing Oso now require installing sqlite? I have no idea how Go dependencies work

Comment (gj) [reply languages/go/go.mod:12]: https://stackoverflow.com/a/66178556

**Review by patrickod — APPROVED** [review]

File: languages/go/go.mod (hunk 1 above)
Comment (gj) [reply languages/go/go.mod]: Believe it no longer will now that I've separated out test dependencies into their own `go.mod`. I used gorm as an example: https://github.com/go-gorm/gorm/blob/e5894ca44951fecc3b3f31f1aa46df7de6024b04/tests/go.mod

**Review by dhatch — APPROVED** [review]
tested, lgtm

**Pull request merged**